### PR TITLE
Onnxruntime nightly install

### PR DESCRIPTION
**Pull request by natke**
Update Python package install instructions. Staged here: https://natke.github.io/onnxruntime/docs/install/

### CODE REVIEW

File: docs/install/index.md [hunk 1]
```diff
@@ -65,31 +91,33 @@ pip install onnxruntime-gpu --extra-index-url https://aiinfra.pkgs.visualstudio.
...
 
 ```bash
-## sklearn
-pip install skl2onnx
+pip install --pre --extra-index-url https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ onnxruntime-qnn
 ```
 
+### Install ONNX Runtime GPU (ROCm)
+
+For ROCm, please follow instructions to install it at the [AMD ROCm install docs](https://rocm.docs.amd.com/projects/install-on-linux/en/docs-6.0.0/). The ROCm execution provider for ONNX Runtime is built and tested with ROCm 6.0.0.
```
Comment (tianleiwu) [inline docs/install/index.md:109]: Our nuget for 1.20 uses 6.2.0.  I suggest to update the link and version number to 6.2.0 here:
https://rocm.docs.amd.com/projects/install-on-linux/en/docs-6.2.0/

File: docs/get-started/with-python.md [hunk 2]
```diff
@@ -258,24 +258,24 @@ If using pip, run `pip install --upgrade pip` prior to downloading.
...
 Before installing nightly package, you will need install dependencies first.
 ```
 python -m pip install coloredlogs flatbuffers numpy packaging protobuf sympy
 ```
 
-Example to install ort-nightly-gpu for CUDA 11.*:
+Example to install onnxruntime-gpu for CUDA 11.*:
 ```
-python -m pip install ort-nightly-gpu --index-url=https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ort-cuda-11-nightly/pypi/simple/
+python -m pip install onnxruntime-gpu --index-url=https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ort-cuda-11-nightly/pypi/simple/
```
Comment (tianleiwu) [inline docs/get-started/with-python.md:273]: Please double check the package name. It used to be ort-nightly-gpu.

Comment (natke) [reply docs/get-started/with-python.md:273]: I checked. This was a change made recently

> pip install --pre --extra-index-url https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ onnxruntime-gpu
Looking in indexes: https://pypi.org/simple, https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/
Collecting onnxruntime-gpu
  Downloading https://aiinfra.pkgs.visualstudio.com/2692857e-05ef-43b4-ba9c-ccf1c22c437c/_packaging/7982ae20-ed19-4a35-a362-a96ac99897b7/pypi/download/onnxruntime-gpu/1.21.dev20241021003/onnxruntime_gpu-1.21.0.dev20241021003-cp310-cp310-win_amd64.whl (279.7 MB)

File: docs/get-started/with-python.md [hunk 3]
```diff
@@ -258,24 +258,24 @@ If using pip, run `pip install --upgrade pip` prior to downloading.
...
 ```
 
-Example to install ort-nightly-gpu for CUDA 12.*:
+Example to install onnxruntime-gpu for CUDA 12.*:
 ```
-python -m pip install ort-nightly-gpu --index-url=https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/
+python -m pip install onnxruntime-gpu --index-url=https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/
```
[the remark this answers is not in the document]
Comment (natke) [reply docs/get-started/with-python.md:278]: See above

File: docs/install/index.md [hunk 4]
```diff
@@ -31,32 +31,58 @@ under [Compatibility](../reference/compatibility).
...
 The default CUDA version for [onnxruntime-gpu in pypi](https://pypi.org/project/onnxruntime-gpu) is 12.x since 1.19.0.
 
 ```bash
 pip install onnxruntime-gpu
 ```
 
+#### Install nightly
+
+```bash
+pip install --pre --extra-index-url https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ onnxruntime-gpu
```
[the remark this answers is not in the document]
Comment (natke) [reply docs/install/index.md:78]: See above

File: docs/install/index.md (hunk 1 above)
Comment (natke) [reply docs/install/index.md]: Fixed

**Review by snnn — DISMISSED** [review]

File: docs/execution-providers/QNN-ExecutionProvider.md [hunk 5]
```diff
@@ -43,7 +43,7 @@ Note: Starting version 1.18.0 , you do not need to separately download and insta
     - Python 3.11.x
     - Numpy 1.25.2 or >= 1.26.4
   - Install: `pip install onnxruntime-qnn`
-  - Install nightly package `python -m pip install -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ort-nightly-qnn`
+  - Install nightly package `python -m pip install -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/onnxruntime-qnn`
```
Comment (snnn) [inline docs/execution-providers/QNN-ExecutionProvider.md:46]: Add --pre

File: docs/get-started/with-python.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (snnn) [reply docs/get-started/with-python.md:278]: Add --pre

File: docs/install/index.md [hunk 6]
```diff
@@ -65,31 +91,30 @@ pip install onnxruntime-gpu --extra-index-url https://aiinfra.pkgs.visualstudio.
...
-```bash
-## tensorflow
-pip install tf2onnx
-```
+#### Install nightly
 
 ```bash
-## sklearn
-pip install skl2onnx
+pip install --pre --extra-index-url https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ onnxruntime-qnn
```
Comment (snnn) [inline docs/install/index.md:104]: Sometimes the doc says "--extra-index-url", sometimes it says "--index-url". Better to keep consistent unless you have a special reason. 
The two args work differently. 
"--extra-index-url" means you will search the packages from two places: pypi.org and the extra one. 
"--index-url" means only one URL will be used.

File: docs/get-started/with-python.md (hunk 2 above)
Comment (tianleiwu) [reply docs/get-started/with-python.md:273]: I see. `--pre` shall be added.

File: docs/execution-providers/QNN-ExecutionProvider.md (hunk 5 above)
Comment (tianleiwu) [reply docs/execution-providers/QNN-ExecutionProvider.md:46]: Missing package name. Also please double check this nightly feed, it is not working in my test:

(py311) >python -m pip install --pre onnxruntime-qnn -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/onnxruntime-qnn
Looking in indexes: https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/onnxruntime-qnn
ERROR: Could not find a version that satisfies the requirement onnxruntime-qnn (from versions: none)
ERROR: No matching distribution found for onnxruntime-qnn

Comment (jywu-msft) [reply docs/execution-providers/QNN-ExecutionProvider.md:46]: index url should be https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ 
e.g.
```
python -m pip install --pre onnxruntime-qnn -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ 
```

File: docs/install/index.md (hunk 6 above)
Comment (natke) [reply docs/install/index.md:104]: Which one do you recommend?

Comment (snnn) [reply docs/install/index.md:104]:
```
pip install --pre --index-url https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ onnxruntime
```
And setup a nightly pipeline to test all such commands for all major platforms:
```
macOS
Windows x86_64
Windows x86_32
Windows ARM64
Linux x86_64
```

Comment (snnn) [reply docs/install/index.md:104]: See: https://github.com/pypa/pip/issues/8606

Comment (natke) [reply docs/install/index.md:104]: Am I right in saying that if you use `--index-url`, you need to install all of the dependencies manually?

File: docs/execution-providers/QNN-ExecutionProvider.md (hunk 5 above)
Comment (natke) [reply docs/execution-providers/QNN-ExecutionProvider.md]: Fixed

File: docs/install/index.md (hunk 6 above)
Comment (snnn) [reply docs/install/index.md:104]: Yes.  --extra-index-url is easier to use.

Comment (snnn) [reply docs/install/index.md:104]: So, we recommend using "--extra-index-url" ?  I think it's fine too.

File: docs/get-started/with-python.md [hunk 7]
```diff
@@ -275,7 +275,7 @@ python -m pip install onnxruntime-gpu --index-url=https://aiinfra.pkgs.visualstu
 
 Example to install onnxruntime-gpu for CUDA 12.*:
 ```
-python -m pip install onnxruntime-gpu --index-url=https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/
+python -m pip install onnxruntime-gpu pre --extra-url=https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/
```
Comment (tianleiwu) [inline docs/get-started/with-python.md:278]: --extra-index-url

File: docs/execution-providers/QNN-ExecutionProvider.md [hunk 8]
```diff
@@ -43,7 +43,7 @@ Note: Starting version 1.18.0 , you do not need to separately download and insta
     - Python 3.11.x
     - Numpy 1.25.2 or >= 1.26.4
   - Install: `pip install onnxruntime-qnn`
-  - Install nightly package `python -m pip install -i https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple/ort-nightly-qnn`
+  - Install nightly package `python -m pip install --pre --extra-index-url https://aiinfra.pkgs.visualstudio.com/PublicPackages/_packaging/ORT-Nightly/pypi/simple`
```
Comment (tianleiwu) [inline docs/execution-providers/QNN-ExecutionProvider.md:46]: missed the package name in pip install command

Comment (natke) [reply docs/execution-providers/QNN-ExecutionProvider.md]: Fixed

**Review by snnn — APPROVED** [review]

**Pull request merged**